src/cata_assert.h

```diff
@@ -0,0 +1,30 @@
+// NOLINTNEXTLINE(cata-header-guard)
+// Due to an inability to supress assert popups when building against mingw-w64 and running on wine
+// We are wrapping the assert macro so that we can substitute functional behavior with that setup
+
+// This copies the semantics of cassert, re-including the file re-defines the macro.
+#undef cata_assert
+
+// Might as well handle NDEBUG at the top level instead of just wrapping one variant.
+#ifdef NDEBUG
+#define cata_assert(_Expression) ((void)0)
+#else
+#ifdef _WIN32
+#include <cstdlib>
+#include <cstdio>
+#define cata_assert(_Expression) \
+    do { \
+        if( _Expression ) { \
+            break; \
+        } \
+        fprintf( stderr, "%s:%d: Assertion `%s` failed.\n", __FILE__, __LINE__, #_Expression ); \
```

Comment:
```suggestion
        fprintf( stderr, "%s at %s:%d: Assertion `%s` failed.\n", __func__, __FILE__, __LINE__, #_Expression ); \
```
From [cppreference](https://en.cppreference.com/w/cpp/error/assert):

> The diagnostic information is required to include the text of expression, as well as the values of the standard macros `__FILE__`, `__LINE__`, and the standard variable `__func__`